File: Stack.cpp
#include<iostream>
using namespace std;
typedef struct node
{
    int data;
    struct node*next;
}NODE,*PNODE;
class Stack
{
    private:
    PNODE first;
    int size;
    public:
    Stack();
    void push(int);
    int pop();
    void Display();
    int Count();

};

Stack::Stack()
{
    first=NULL;
    size=0;
}

void Stack::push(int no)
{
    PNODE newn=new NODE;
    newn->data=no;
    newn->next=NULL;

    if(first==NULL)
    {
        first=newn;
    }
    else
    {
        newn->next=first;
        first=newn;
    }
    size++;
}

int Stack::pop()
{
    PNODE temp=first;
    int no=0;
    if(first==NULL)
    {
        cout<<"Stack is empty";
        return -1;
    }
    else if(first->next==NULL)
    {
        no=first->data;
        delete first;
        first=NULL;
    }
    else
    {
        no=first->data;
        first=first->next;
        delete temp;
    }
    size--;
    return no;
}
void Stack::Display()
{
    PNODE temp=first;
    for(int i=1;i<=size;i++)
    {
        cout<<"|"<<temp->data<<"|"<<"\n";
        temp=temp->next;
    }
    cout<<"\n";
}
int Stack::Count()
{
    return size;
}

int main()
{
    Stack obj;
    obj.push(51);
    obj.push(21);
    obj.push(11);
    obj.push(8);
    obj.Display();
    int iRet=obj.Count();
    cout<<"Count Are:"<<iRet<<"\n";
    int Ret=obj.pop();
    obj.Display();
    cout<<"Poped element are:"<<Ret<<"\n";
    return 0;
}

File: Stack.java
import java.net.PortUnreachableException;

import javax.swing.plaf.synth.SynthUI;

class Node
{
    public int data;
    public Node next;
}
class Insert
{
    private Node first;
    private int size;
    public Insert()
    {
        first=null;
        size=0;
    }

    public void Push(int no)
    {
        Node newn=new Node();
        newn.data=no;
        newn.next=null;
        if(first==null)
        {
            first=newn;
        }
        else
        {
            newn.next=first;
            first=newn;
        }
        this.size++;
    }
    public int Pop()
    {
        int no=0;
        if(first==null)
        {
            return-1;
        }
        else if(size==1)
        {
            first=null;
        }
        else
        {
            
            
            no=first.data;
            first=first.next;
            
        }
        this.size--;
        return no;
    }
    public void Display()
    {
        Node temp=first;
        while(temp!=null)
        {
            System.out.print("|"+temp.data+"|->");
            temp=temp.next;
        }
        System.out.print("null");
    }
    public int Count()
    {
        return size;
    }
}
class Stack 
{
    public static void main(String arg[])
    {
        Insert obj=new Insert();
        obj.Push(51);
        obj.Push(21);
        obj.Push(11);
        obj.Push(8);
        int iRet=obj.Pop();
        System.out.println("Poped data are:"+iRet);
        obj.Display();
        int iret=obj.Count();
        System.out.println("\nCount are:"+obj.Count());
    }
}
